gtsfm/utils/io.py

```diff
@@ -418,3 +420,17 @@ def save_track_visualizations(
         stacked_image = image_utils.vstack_image_list(patches)
         save_fpath = os.path.join(save_dir, f"track_{i}.jpg")
         save_image(stacked_image, img_path=save_fpath)
+
+
+def read_from_compressed_file(file_path: Path) -> Optional[Any]:
+    """Reads data using pickle from a compressed file, if it exists."""
+    if file_path.exists():
+        return pickle.load(BZ2File(file_path, "rb"))
+
+    return None
+
+
+def write_to_compressed_file(data: Any, file_path: Path) -> None:
```

Comment:
`compressed_file` is a bit ambiguous, since there are many types of file compression.

Should we just call it `write_to_compressed_file_bz2` or `write_file_bz2` or `write_file_pkl()`? @akshay-krishnan what are your thoughts?

Reply:
I am fine with these names. I think `write_to_bz2_file` and `read_to_bz2_file` works. What do you think?

Reply:
yeah I like that. thanks for updating it. marking as resolved.